.github/workflows/deploy-docker-compose.yml

```diff
@@ -0,0 +1,161 @@
+            https://ghcr.io/v2/${IMAGE_NAME}/tags/list \
+            | jq -r --arg TAG "${IMAGE_TAG}" '.tags[] | select(. == $TAG)')
+          
+          if [ -z "$TAG_EXISTS" ]; then
+            echo "Image ${IMAGE_NAME}:${IMAGE_TAG} does not exist."
+            exit 1
+          else
+            echo "Image ${IMAGE_NAME}:${IMAGE_TAG} exists."
+          fi
+
+  deploy:
+    needs: prepare-deploy
+    runs-on: ubuntu-latest
+    environment: 
+      name: ${{ inputs.environment }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: SSH to VM and execute docker compose down (if exists)
+        uses: appleboy/ssh-action@v1.0.3
+        with:
+          host: ${{ vars.VM_HOST }}
+          username: ${{ vars.VM_USERNAME }}
+          key: ${{ secrets.VM_SSH_PRIVATE_KEY }}
+          proxy_host: ${{ vars.DEPLOYMENT_GATEWAY_HOST }}
+          proxy_username: ${{ vars.DEPLOYMENT_GATEWAY_USER }}
+          proxy_key: ${{ secrets.DEPLOYMENT_GATEWAY_SSH_KEY }}
+          proxy_port: ${{ vars.DEPLOYMENT_GATEWAY_PORT }}
+          script: |
+            #!/bin/bash
+            set -e  # Exit immediately if a command exits with a non-zero status
+
+            COMPOSE_FILE="${{ inputs.docker-compose-file }}"
+
+            # Check if docker-compose.prod.yml exists
+            if [ -f "$COMPOSE_FILE" ]; then
+              echo "$COMPOSE_FILE found."
+          
+              # Check if .env exists
+              if [ -f ".env" ]; then
+                docker compose -f "$COMPOSE_FILE" --env-file=".env" down --remove-orphans --rmi all
+              else
+                docker compose -f "$COMPOSE_FILE" down --remove-orphans --rmi all
+              fi
+            else
+              echo "$COMPOSE_FILE does not exist. Skipping docker compose down."
+            fi
+
+      - name: Copy docker compose file to VM host
+        uses: appleboy/scp-action@v0.1.7
+        with:
+          host: ${{ vars.VM_HOST }}
+          username: ${{ vars.VM_USERNAME }}
+          key: ${{ secrets.VM_SSH_PRIVATE_KEY }}
+          proxy_host: ${{ vars.DEPLOYMENT_GATEWAY_HOST }}
+          proxy_username: ${{ vars.DEPLOYMENT_GATEWAY_USER }}
+          proxy_key: ${{ secrets.DEPLOYMENT_GATEWAY_SSH_KEY }}
+          proxy_port: ${{ vars.DEPLOYMENT_GATEWAY_PORT }}
+          source: ${{ inputs.docker-compose-file }}
+          target: /home/${{ vars.VM_USERNAME }}
+
+      - name: SSH to VM and create .env file
+        uses: appleboy/ssh-action@v1.0.3
+        with:
+          host: ${{ vars.VM_HOST }}
+          username: ${{ vars.VM_USERNAME }}
+          key: ${{ secrets.VM_SSH_PRIVATE_KEY }}
+          proxy_host: ${{ vars.DEPLOYMENT_GATEWAY_HOST }}
+          proxy_username: ${{ vars.DEPLOYMENT_GATEWAY_USER }}
+          proxy_key: ${{ secrets.DEPLOYMENT_GATEWAY_SSH_KEY }}
+          proxy_port: ${{ vars.DEPLOYMENT_GATEWAY_PORT }}
+          script: |
+            touch .env
```

Comment:
I have seen other projects using different naming like `.env`, `.env.prod`, etc
So maybe we can also add another input like `environment_file_name`